### PR TITLE
refactor: migrate TagNode and SubTags pagination to CursorPaginationManager

### LINKED ISSUE #5141: Refactor Tag & Request Pagination Components
Complete the core refactoring by applying `CursorPaginationManager` to tag-related components and the Requests screen, including integration with `react-infinite-scroll-component`.

**Parent Issue:** 
- #5100

**Prerequisite Issue:** 
- #5143

### Pull Requests

- Submit no more than 10 code files per PR. This will make reviews easier and speed up resolution of the issue.

### Files to Modify
- `src/components/TagActions/Node/TagNode.tsx` - Replace `fetchMoreSubTags`
- `src/components/TagActions/Node/TagNode.spec.tsx` - Update tests
- `src/screens/SubTags/SubTags.tsx` - Replace `loadMoreSubTags`
- `src/screens/SubTags/SubTags.spec.tsx` - Update tests
- `src/screens/Requests/Requests.tsx` - Integrate with InfiniteScroll
- `src/screens/Requests/Requests.spec.tsx` - Update tests

### TagNode Component Refactoring
- [ ] Remove `fetchMoreSubTags` manual implementation (lines 70-96)
- [ ] Remove nested `updateQuery` logic
- [ ] Replace with `CursorPaginationManager` component
- [ ] Maintain subtag display and interaction functionality
- [ ] **Code reduction: ~30 lines removed**

### SubTags Screen Refactoring
- [ ] Remove `loadMoreSubTags` pattern
- [ ] Replace with `CursorPaginationManager` component
- [ ] Maintain existing tag management functionality
- [ ] Consistent with other tag components
- [ ] **Code reduction: ~25 lines removed**

### Requests Screen Refactoring
- [ ] Integrate `CursorPaginationManager` with `react-infinite-scroll-component`
- [ ] Remove `loadMoreRequests` manual implementation
- [ ] Maintain infinite scroll UX
- [ ] Handle scroll events correctly
- [ ] **Special consideration:** Ensure compatibility with InfiniteScroll component
- [ ] **Code reduction: ~30 lines removed**

### Testing Updates
- [ ] TagNode tests updated for new structure
- [ ] SubTags tests updated
- [ ] Requests tests updated with InfiniteScroll integration
- [ ] Test infinite scroll functionality (check for page-2 items like "User9 Test")
- [ ] Test pagination edge cases
- [ ] All existing tests pass
- [ ] Maintain or improve test coverage

### InfiniteScroll Integration Pattern
- [ ] Document pattern for using CursorPaginationManager with InfiniteScroll
- [ ] Ensure resilient scroll testing approach (query container, fallback to window)
- [ ] No scroll-related test flakiness

## 🔗 Dependencies

- **Requires:** Issue for PR #1 (CursorPaginationManager component) must be completed first

## ⏱️ Estimated Effort

**6-8 hours**

## 🎯 Success Metrics

- Tag-related components use standardized pagination
- Requests screen maintains smooth infinite scroll UX
- **Code Reduction:** 85+ lines of pagination logic removed
- All tests passing with resilient scroll testing
- No scroll-related test flakiness

### PR DESCRIPTION
**Pull request by abhishekblue**
**What kind of change does this PR introduce?**
Refactor

**Issue Number:**
Fixes #5141

**Summary**

- Refactored `TagNode` and `SubTags` to use `CursorPaginationManager`
- Replaced `DataGridWrapper` in `SubTags` with flex-based row layout consistent with other migrated screens
- Updated all related tests and mocks; fixed pre-existing design token violations in SubTags CSS

**Does this PR introduce a breaking change?**
No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**
Yes

## Summary by CodeRabbit

* **New Features**
  * Button-driven pagination for loading additional subtags and improved keyboard activation (Enter/Space) on tag rows.

* **Bug Fixes**
  * More consistent subtag loading, clearer empty/error displays, and stable expand/collapse behavior with the new pagination flow.

* **Style**
  * Shadows converted to theme tokens, new layout utility classes, and an added color token for shadow opacity.

* **Tests**
  * Test suite simplified and updated to reflect the button-based pagination and accessibility changes.

### CODE REVIEW

**Comment by abhishekblue** [comment]
@coderabbitai Full in-depth review.

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where test cleanup is important and response times are highly variable. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

**Comment by palisadoes** [comment]
Please address these findings:
- https://github.com/PalisadoesFoundation/talawa-admin/pull/7510#issuecomment-4103416176

**Comment by abhishekblue** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where test cleanup is important and response times are highly variable. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

**Comment by abhishekblue** [comment]
@coderabbitai Requests screen is skipped because it does not use cursor pagination, one uses offset (skip/first), the other fetches everything at once.

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where test cleanup is important and response times are highly variable. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

File: src/style/tokens/colors.css [hunk 1]
```diff
@@ -40,6 +40,7 @@
   --color-black-15: rgba(0, 0, 0, 0.15);
   --color-black-20: rgba(0, 0, 0, 0.2);
   --color-black-25: rgba(0, 0, 0, 0.25);
+  --color-black-30: rgba(0, 0, 0, 0.3);
```
Comment (palisadoes) [inline src/style/tokens/colors.css]: Use an existing color and revert this change.

**Review by palisadoes — CHANGES_REQUESTED** [review]
Use an existing color

**Comment by palisadoes** [comment]
@coderabbitai full review and reevaluate all your previous comments that may have been dismissed, ignored or resolved. Evaluate whether the goals of the issue linked in the first PR comment are met. If there are test cases, ensure that the updated or created tests won't be flaky, especially in a sharded testing environment where test cleanup is important and response times are highly variable. Post your conclusion as an easily readable comment that also includes your findings outside the diffs.

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please address this error that was found:
- https://github.com/PalisadoesFoundation/talawa-admin/pull/7510#issuecomment-4154637119

**Comment by abhishekblue** [comment]
> Please address this error that was found:
> 
> * [#7510 (comment)](https://github.com/PalisadoesFoundation/talawa-admin/pull/7510#issuecomment-4154637119)

@coderabbitai @palisadoes 
Coderabbit is looking at stale diff for some reason. I am already using `--color-black-25 (rgba(0,0,0,0.25)` and removed  `--color-black-30`.

**Comment by palisadoes** [comment]
Please fix the failing tests
